File: src/main/java/me/nallar/jaropener/JarOpener.java
package me.nallar.jaropener;

import me.nallar.javatransformer.api.AccessFlags;
import me.nallar.javatransformer.api.JavaTransformer;

import java.io.*;

public class JarOpener {
	public static void main(String[] args) {
		if (args.length != 2) {
			System.err.println("Usage: java -jar jaropener.jar input.jar output.jar");
			return;
		}

		JavaTransformer transformer = new JavaTransformer();
		transformer.addTransformer((x) -> {
			x.accessFlags(JarOpener::open);
			x.getMethods().forEach((it) -> it.accessFlags(JarOpener::open));
			x.getFields().forEach((it) -> it.accessFlags(JarOpener::open));
		});

		transformer.transformJar(new File(args[0]), new File(args[1]));
	}

	private static AccessFlags open(AccessFlags flag) {
		return flag.without(AccessFlags.ACC_FINAL).makeAccessible(true);
	}
}

File: src/main/kotlin/me/nallar/JarOpener.kt
@file:JvmName("JarOpener")

package me.nallar

import me.nallar.javatransformer.api.AccessFlags
import me.nallar.javatransformer.api.JavaTransformer
import me.nallar.javatransformer.api.TransformationException
import java.nio.file.Paths

fun main(args: Array<String>) {
	if (args.size != 2) {
		System.err.println("Usage: java -jar jaropener.jar input.jar output.jar")
		return
	}

	val transformer = JavaTransformer()
	transformer.addTransformer({
		try {
			it.accessFlags({ open(it) })
			it.members.forEach({ it.accessFlags({ open(it) }) })
		} catch (e: TransformationException) {
			System.err.println("Error transforming class: " + it.name)
			e.printStackTrace()
		}
	})

	transformer.transform(Paths.get(args[0]), Paths.get(args[1]))
}

private fun open(flag: AccessFlags): AccessFlags {
	return flag.without(AccessFlags.ACC_FINAL).makeAccessible(true)
}
